src/ansys/dpf/composites/_composite_model_impl.py

```diff
@@ -0,0 +1,613 @@
+# New interface after 2023 R2
+from typing import Any, Callable, Collection, Dict, List, Optional, Sequence, cast
+from warnings import warn
+
+import ansys.dpf.core as dpf
+from ansys.dpf.core import FieldsContainer, MeshedRegion, Operator, UnitSystem
+from ansys.dpf.core.server_types import BaseServer
+import numpy as np
+from numpy.typing import NDArray
+
+from .composite_scope import CompositeScope
+from .data_sources import (
+    CompositeDataSources,
+    ContinuousFiberCompositesFiles,
+    get_composites_data_sources,
+)
+from .failure_criteria import CombinedFailureCriterion
+from .layup_info import (
+    ElementInfo,
+    LayerProperty,
+    LayupPropertiesProvider,
+    add_layup_info_to_mesh,
+    get_element_info_provider,
+)
+from .layup_info.material_operators import MaterialOperators, get_material_operators
+from .layup_info.material_properties import MaterialProperty, get_constant_property_dict
+from .result_definition import FailureMeasureEnum
+from .sampling_point import SamplingPointNew
+from .server_helpers import upload_continuous_fiber_composite_files_to_server
+from .unit_system import get_unit_system
+
+
+def _deprecated_composite_definition_label(func: Callable[..., Any]) -> Any:
+    """
+    Can be used to mark composite_definition_label as deprecated.
+
+    It will result in a warning being emitted when the function is used.
+    """
+    function_arg = "composite_definition_label"
+
+    def inner(*args: Sequence[Any], **kwargs: Sequence[Any]) -> Any:
+        if function_arg in kwargs.keys():
+            if kwargs[function_arg]:
+                warn(
+                    f"Use of {function_arg} is deprecated. Function {func.__name__}."
+                    " can be called without this argument.",
+                    category=DeprecationWarning,
+                    stacklevel=2,
+                )
+        return func(*args, **kwargs)
+
+    return inner
+
+
+class CompositeModelImpl:
+    """Provides access to the basic composite postprocessing functionality.
+
+    This class supports DPF Server version 7.0 (2024 R1) and later.
+    On initialization, the ``CompositeModel`` class automatically adds composite lay-up
+    information to the meshed regions. It prepares the providers for different lay-up properties
+    so that they can be efficiently evaluated. The composite_files provided are automatically
+    uploaded to the server if needed.
+
+    .. note::
+
+        When creating a ``CompositeModel`` instance, several providers are created and
+        lay-up information is added to the DPF meshed regions. Depending on the use
+        case, it can be more efficient to create the providers separately.
+
+    Parameters
+    ----------
+    composite_files:
+        Use the :func:`.get_composite_files_from_workbench_result_folder` function to obtain
+        the :class:`.ContinuousFiberCompositesFiles` object.
+    server:
+        DPF Server on which the model is created
```

Comment:
```suggestion
        DPF Server to create the model on.
```
We allow ending a sentence with a preposition as this is how people speak. Plus, it's easier to understand.